### PR TITLE
3437 remove redundant return from healthchecker

### PR DESCRIPTION
**Pull request by akaspin**

### CODE REVIEW

File: healthcheck.go [hunk 1]
```diff
@@ -12,13 +12,13 @@ import (
 
 // HealthCheckable is a subject which's health can be checked
 type HealthCheckable interface {
-	Healthy() (bool, error)
+	Healthy() error
 }
 
 // CheckHealth can be used to wrap functions so they fulfill the HealthCheckable interface
-type CheckHealth func() (bool, error)
+type CheckHealth func() error
```
Comment (mbertschler) [inline healthcheck.go]: Do we need `HealthCheckable interface` and `CheckHealth func`? This looks complicated to me.
Then `AddCheck` can be:
```diff
- func (h *HealthChecker) AddCheck(name string, checkable HealthCheckable) {
+ func (h *HealthChecker) AddCheck(name string, checkable func() error) {
```
And use any function with an error as check.

Comment (akaspin) [reply healthcheck.go]: @mbertschler It's code from @dynamix big PR. Where in turn current refactored implementation. All is discoursed already.

Comment (mbertschler) [reply healthcheck.go]: Do you mean the "refactor to use dependancy injection / cleanup" PR?

Comment (akaspin) [reply healthcheck.go]: Yep

Comment (dynamix) [reply healthcheck.go]: @mbertschler `CheckHealth` is just a helper to make it easy to wrap trivial cases and conform to the interface. While I agree that we could remove the HealthCheckable interface - objects reporting on their health would likely diverge (people will have different method names, pass unnamed functions and so on). I'm ambivalent - both things are fine for me.

**Pull request merged**